chainstate/test-suite/src/tests/double_spend_tests.rs

```diff
@@ -426,10 +424,7 @@ fn duplicate_tx_in_the_same_block(#[case] seed: Seed) {
             tf.process_block(block, BlockSource::Local).unwrap_err(),
             ChainstateError::ProcessBlockError(BlockError::CheckBlockFailed(
                 CheckBlockError::CheckTransactionFailed(
-                    CheckBlockTransactionsError::DuplicatedTransactionInBlock(
-                        second_tx_id,
-                        block_id
-                    )
+                    CheckBlockTransactionsError::DuplicateInputInBlock(block_id)
```

Comment:
At this point I'm not sure if error `DuplicatedTransactionInBlock` is reachable. Because every duplicated tx will have duplicated inputs that will be reported. The case with empty inputs (that gives the same tx id) is handled separately.

I left the check for duplicated txs untouched just as a safe net, but I'd like to know your opinions whether it is still valid.

Reply:
Usually the right way to solve such problems is to write unit tests that prove whatever you're trying to prove, instead of adding blanket checks that may or may not be reached.

Reply:
That's the problem - I can't figure out the case how to create duplicated transaction that would trigger `DuplicatedTransactionInBlock ` from `check_transactions`

Reply:
Well, then we can remove the duplicated transactions check and just test for it in unit tests.